### PR TITLE
Improve and fix dockerode types

### PR DESCRIPTION
**Pull request by samuel-olivier**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Docker API](https://docs.docker.com/engine/api/v1.41) (more relevant links to the doc in each commit description)

### CODE REVIEW

**Comment by samuel-olivier** [comment]
Friendly reminder @Seikho @nlaplante @isac322 @LazarusX @meisenzahl @thegecko @CameronDiver @P4sca1 @stuartthomson @userlerueda

**Comment by samuel-olivier** [comment]
Anyone can review this MR please ? :)

@Seikho @nlaplante @isac322 @LazarusX @meisenzahl @thegecko @CameronDiver @P4sca1 @stuartthomson @userlerueda

**Comment by thegecko** [comment]
There's quite a bit of code churn for formatting, so it's difficult to trivially see the changes. Recommend this is split into 2 PRs, with the formatting separate to help with reviews.

**Comment by Seikho** [comment]
It looks like it'll be a breaking change due to `RestartPolicy` being split into `HostRestartPolicy` and `TaskRestartPolicy`, but it seems necessary to split it up. Apart from the questionable formatting style (:stuck_out_tongue:) it looks okay to me.

**Review by Seikho — APPROVED** [review]

**Comment by samuel-olivier** [comment]
Thanks for the review @thegecko @Seikho

@thegecko I split the PR in several commits to make the review process easier. All the formatting is done in one commit (3a12557cd6e6a413f6991965d578565782a6292d) and each other commit contains one fix or one improvement. Reviewing commit by commit seems way safer to avoid missing some changes.

@Seikho I agree the formatting style is questionnable (I wouldn't have choose those prettier settings either) but at least now it matches the config from the repo which seems better to me than having formatting inconsistencies in the code 🙂

File: types/dockerode/index.d.ts [hunk 1]
```diff
@@ -1428,19 +1473,9 @@ declare namespace Dockerode {
...
-
     interface ServiceListOptions {
-        filters: {
-            id?: string[] | undefined;
-            label?: string[] | undefined;
-            mode?: Array<'replicated' | 'global'> | undefined;
-            name?: string[] | undefined;
-        };
-
+        filters?: string | undefined;
```
Comment (Kirween) [inline types/dockerode/index.d.ts:1477]: this version break working project, dockerode accept both declaration, this should be 
 ```
 filters?: string | {
            id?: string[] | undefined;
            label?: string[] | undefined;
            mode?: Array<'replicated' | 'global'> | undefined;
            name?: string[] | undefined;
        } | undefined;
```

Comment (samuel-olivier) [reply types/dockerode/index.d.ts:1477]: Indeed, thank you ! Here is the fix: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63263

Comment (Kirween) [reply types/dockerode/index.d.ts:1477]: > fix has been published in [version 3.3.14](https://www.npmjs.com/package/@types/dockerode/v/3.3.14)

Thank you ! It's good now

**Pull request merged**